packages/grafana-ui/src/components/Table/TableNG/utils.test.ts

```diff
@@ -0,0 +1,1731 @@
+    it('should create an emotion css class', () => {
+      const styles = getFooterStyles('flex-start');
+
+      // Check that the footerCell style has been created
+      expect(styles.footerCell).toBeDefined();
+
+      // Get the CSS string representation
+      const cssString = styles.footerCell.toString();
+
+      // Verify it's an Emotion CSS class
+      expect(cssString).toContain('css-');
+    });
+
+    it('should use the provided justification value', () => {
+      const styles = getFooterStyles('center');
+
+      // Create a DOM element and apply the CSS class
+      document.body.innerHTML = `<div class="${styles.footerCell}">Test</div>`;
+      const element = document.querySelector('div');
+
+      // Get the computed style
+      const computedStyle = window.getComputedStyle(element!);
+
+      // Check the CSS property
+      expect(computedStyle.justifyContent).toBe('center');
+    });
+
+    it('should default to space-between when no justification is provided', () => {
+      const styles = getFooterStyles(undefined as any);
+
+      // Create a DOM element and apply the CSS class
+      document.body.innerHTML = `<div class="${styles.footerCell}">Test</div>`;
+      const element = document.querySelector('div');
+
+      // Get the computed style
+      const computedStyle = window.getComputedStyle(element!);
+
+      // Check the CSS property
+      expect(computedStyle.justifyContent).toBe('space-between');
+    });
+
+    // Clean up after all tests
+    afterAll(() => {
+      document.body.innerHTML = '';
+    });
+  });
+
+  describe('extractPixelValue', () => {
+    it('should extract numeric value from pixel string', () => {
+      expect(extractPixelValue('100px')).toBe(100);
+      expect(extractPixelValue('42px')).toBe(42);
+      expect(extractPixelValue('0px')).toBe(0);
+    });
+
+    it('should handle numeric input', () => {
+      expect(extractPixelValue(100)).toBe(100);
+      expect(extractPixelValue(42)).toBe(42);
+      expect(extractPixelValue(0)).toBe(0);
+    });
+
+    it('should handle string numbers without units', () => {
+      expect(extractPixelValue('100')).toBe(100);
+      expect(extractPixelValue('42')).toBe(42);
+      expect(extractPixelValue('0')).toBe(0);
+    });
+
+    it('should handle decimal values', () => {
+      expect(extractPixelValue('100.5px')).toBe(100.5);
+      expect(extractPixelValue('42.75px')).toBe(42.75);
+      expect(extractPixelValue(100.5)).toBe(100.5);
+    });
+
+    it('should handle negative values', () => {
+      expect(extractPixelValue('-100px')).toBe(-100);
+      expect(extractPixelValue('-42px')).toBe(-42);
+      expect(extractPixelValue(-100)).toBe(-100);
+    });
+
+    it('should handle other CSS units by removing them', () => {
+      expect(extractPixelValue('100em')).toBe(100);
+      expect(extractPixelValue('42rem')).toBe(42);
+      expect(extractPixelValue('10vh')).toBe(10);
+      expect(extractPixelValue('20vw')).toBe(20);
```

Comment:
Should we remove them or return 0? Question for the function itself. Not a big deal.